### PR TITLE
[Forge] Use poetry in GH workflows

### PR DESCRIPTION
**Pull request by sionescu**
### Description

Use poetry to fetch Forge wrapper deps in GH workflows.
Keep the tasks that runs "pip install" to be removed in the future.

### CODE REVIEW

**Review by geekflyer — APPROVED** [review]

File: .github/workflows/forge-gcp.yaml [hunk 1]
```diff
@@ -60,13 +60,19 @@ jobs:
 
       - uses: actions/setup-python@13ae5bb136fac2878aff31522b9efb785519f984 # pin@v4
 
-      - name: Install python deps
```
Comment (perryjrandall) [inline .github/workflows/forge-gcp.yaml:63]: This is not backwards compatible.... they both need to exist or be cherry picked in

**Review by perryjrandall — CHANGES_REQUESTED** [review]
These changes (removing pip from workflows) are not backwards compatible

This is probably going to break all branches

Just add poetry for now and point all new things to poetry, then we can separately remove pip and make sure its picked into all branches (after giving time for pull requests to rebase)

Also might wanna give folks a heads up that they potentially need to rebase

File: .github/workflows/forge-gcp.yaml (hunk 1 above)
Comment (sionescu) [reply .github/workflows/forge-gcp.yaml:63]: Do you mean cherry-picked in the release branches ?

**Comment by sionescu** [comment]
@perryjrandall done.

**Comment by perryjrandall** [comment]
If those new shell files (find images.sh determinator.sh) dont exist then they're probably going to break all the old branches they're also not backwards compatible

**Review by rustielin — APPROVED** [review]

**Pull request closed without merging**